src/cargo/core/source_id.rs

```diff
@@ -752,6 +752,20 @@ impl PartialEq for SourceIdInner {
     }
 }
 
+impl SourceKind {
+    pub fn protocol(&self) -> Option<&str> {
```

Comment:
Do we need to `pub` this?

It look like a implementation details for `Display` impl, especially that `SparseRegistry` return nothing here.

Reply:
Huh, not sure why I made it `pub`.  Switched to `pub(crate)`.